pkg/transport/requester.go

```diff
@@ -0,0 +1,22 @@
+package transport
+
+import (
+	"context"
+
+	"github.com/cloudevents/sdk-go/pkg/binding"
+)
+
+// Requester sends a message and receives a response
+//
+// Optional interface that may be implemented by protocols that support
+// request/response correlation.
+type Requester interface {
+	// Request sends m like Sender.Send() but also arranges to receive a response.
```

Comment:
Why did you removed the constraint on implementing Sender?

Reply:
You might want to only support request and not send